lstebner/growler
# rebuild without jQuery

## description

this PR rebuilds the "growl alert plugin" without jQuery, and using only vanilla javascript. in order to be able to add to unit tests, and a demo page, it became required to add modern build tools as well so webpack, and babel have been introduced along with jest and testing-library. 

## details

- adds .gitignore
- rewritten README in markdown
- babel.config setup
- `src` and `dist` directories
- jest setup and config
- first package.json
- `Growler` function ("class")
- modular CSS using CSS variables
- `Growler` unit tests
- demo of functionality
- webpack configuration
- deletes all code from the original plugin

## how to test this PR

you can check out this branch and run `yarn test`, and also `yarn develop` to see things in action. 

## what else needs done?

the original version of this was hardly full featured, but this one isn't really ready for prime time use either. more work to come to clean this up and turn it into a packagable module. Some things in mind are auto dismiss as an option, improving the demo, getting the demo out of the index, adding click to hide functionality, and cleaning up the code to separate the  container from the message view code, and the event handling logic. probably more by the time i get there!

## eye candy

![image](https://user-images.githubusercontent.com/628757/127787253-822384c9-b949-483a-81fd-b11f81609b11.png)


![Screen Shot 2021-08-01 at 3 02 28 PM](https://user-images.githubusercontent.com/628757/127787242-5e7d8c80-bf2e-4207-8cd4-bb82c5440177.png)
